src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,119 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int STOCK_ARR_LENGTH = 16;
+    private static final float MAX_FILL = 0.75f;
+    private static final int GROWTH_COEFICIENT = 2;
+    private Node<K,V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = (Node<K, V>[]) new Node[STOCK_ARR_LENGTH];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int pos = bucketFor(key);
+        if (table[pos] == null) {
+            increaseSize();
+            table[pos] = new Node<>(key, value);
+            return;
+        }
+        Node<K,V> tempNode = closestNode(key);
+        if (keysAreEqual(key, tempNode.key)) {
+            tempNode.value = value;
+            return;
+        }
+        tempNode.next = new Node<>(key, value);
+        increaseSize();
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K,V> tempNode = closestNode(key);
+        return tempNode != null
+                && tempNode.hash == hashOf(key)
+                && (keysAreEqual(tempNode.key, key))
+                ? tempNode.value : null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int bucketFor(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode()) % table.length;
+    }
+
+    private Node<K,V> closestNode(K key) {
+        Node<K,V> currNode = table[bucketFor(key)];
+        if (currNode == null) {
+            return null;
+        }
+        while (currNode.next != null) {
+            if (currNode.hash == hashOf(key) && keysAreEqual(currNode.key, key)) {
+                return currNode;
+            }
+            currNode = currNode.next;
+        }
+        return currNode;
+    }
+
+    private void increaseSize() {
+        if (++size >= table.length * MAX_FILL) {
+            growArr();
+        }
+    }
+
+    private void growArr() {
+        Node<K,V>[] oldArrCash = table;
+        table = (Node<K, V>[])new Node[table.length * GROWTH_COEFICIENT];
+        size = 0;
```

Comment:
what is the reason to assign '0' and then increment it in addToEndOfBucket() method?